cilium/CFP-25694-move-cilium-cli-to-cilium-repo.md

```diff
@@ -0,0 +1,145 @@
+# CFP-25694: Move cilium/cilium-cli code into cilium/cilium repository
+
+**Code Name: Mega Ketchup**
+
+**SIG: [sig-cilium-cli]**
+
+**Begin Design Discussion:** 2023-05-25
+
+**Cilium Release:** 1.15
+
+**Authors:** Michi Mutsuzaki `<michi@isovalent.com>`
+
+## Summary
+
+Move [cilium/cilium-cli] code into [cilium/cilium] repository.
+
+## Motivation
+
+The main motivation for moving [cilium/cilium-cli] code into [cilium/cilium]
+repository is to minimize the feature development overhead for adding new tests
+to `cilium connectivity test`.
+
+Currently, adding a new test to `cilium connectivity test` requires the following
+steps:
+
+- Open a pull request against [cilium/cilium-cli] repo to add a new test, and
+  wait for the CI build to finish.
+- Modify GitHub workflows files to pick up the CI version of `cilium-cli` in
+  your pull request against [cilium/cilium] repo.
+- Once the new tests passes, undo the change in the workflow files.
+- Get these pull requests reveiewed and merged.
+
+Even after all these steps, the new test does not become a part of the regular
+CI until the [sig-cilium-cli] team releases a new version of `cilium-cli` that
+includes the test, and Renovate updates GitHub wokflow files to pick up the new
+`cilium-cli` version. This has been a major source of frustration for some Cilium
+contributors.
+
+## Goals
+
+* Move [cilium/cilium-cli] code into [cilium/cilium] main branch under `cli/`
+  directory. This enables Cilium contributors to implement new features with
+  tests in a single pull request against [cilium/cilium] repository.
+* Establish a backporting process for changes inside `cli/` directory.
```

Comment:
there are more details in "Impact: Backporting to stable branches" section. let me know if it makes sense 👀

Reply:
Ah, I missed the `since` in the middle there. I think the _goal_ in that case is "Continue to test older Cilium versions against new versions of the CLI". The extra backporting overhead is a logical consequence of the proposal.

Reply:
> then provide a way for cilium/cilium PRs to point to those CLI dev images?

I added this - https://github.com/cilium/cilium/pull/25228. However, bunch of workflows `# pull_request: {}` need to be uncommented though to make changes effective.